File: cpp/excel_sheet_column_title.cpp
168. Excel Sheet Column Title 
Given a positive integer, return its corresponding column title as appear in an Excel sheet.

For example:

    1 -> A
    2 -> B
    3 -> C
    ...
    26 -> Z
    27 -> AA
    28 -> AB 

class Solution {
public:
    string convertToTitle(int n) {
        string res;
        while (n>0) {
            int num=n%26;
            if (num==0) {
                // borrow from high digit
                n-=26;
                num=26;
            }
            res.push_back('A'+num-1);
            n/=26;
        }
        
        reverse(res.begin(), res.end());
        return res;
    }
};

File: java/excel_sheet_column_title.java
public class Solution {
    public String convertToTitle(int n) {
        StringBuilder sb=new StringBuilder();
        while (n>0) {
            int num=n%26;
            if (num==0) {
                n-=26;
                num=26;
            }
            sb.append((char)(num+'A'-1));
            n/=26;
        }
        sb.reverse();
        return sb.toString();
    }
}
